shared/js/blobview.js

```diff
@@ -6,6 +6,15 @@ var BlobView = (function() {
     throw Error(msg);
   }
 
+  var decoderCache = {};
+  function getDecoder(encoding) {
+    if (encoding in decoderCache) {
+      return decoderCache[encoding];
+    }
+    var decoder = decoderCache[encoding] = new TextDecoder(encoding);
+    return decoder;
+  }
```

Comment:
There's definitely some performance overhead to creating lots of `TextDecoder`s. I tried without the cache, and scanning my music collection went from 35 seconds to 45 seconds.